### ISSUE TITLE
[SG2] libcamera not recognizing camera

### ISSUE THREAD

**Issue by gitwah** [linux-surface/linux-surface#555]
I've installed libcamera and the instructions provided [here](https://github.com/linux-surface/linux-surface/wiki/Camera-Support) in order to get the front cam of my Surface Go 2 running. 
It isn't recognized yet, though, and I don't really know why. I'd appreciate it if somebody would help me troubleshoot this issue. This would provide an opportunity to update the wiki, too, as the instructions aren't fully foolproof (as proven by me).

### Environment

- Hardware model: Surface Go 2
- Kernel version: 5.13.13-arch1-1
- Distribution: Arch
 
<details>
I've built libcamera from git and configured it according to the wiki:

`$ CC=clang CXX=clang++ meson build -Dpipelines=uvcvideo,ipu3 -Dprefix=/usr
$ ninja -C build
$ sudo ninja -C build install
`

The cam is not recognized by `cam --list`:
`[0:14:49.856131660] [4938]  INFO Camera camera_manager.cpp:294 libcamera v0.0.0+2983-3335d5a5
Available cameras:
`
using `qcam` results in a GUI popup asking for input on which camera to use.

(I have added `acpi_enforce_resources=lax` to the systemd-boot options and also made sure that $USER is part of the video group)  
</details>

Could somebody give me some pointers on what logs to check? Thank you!

**Comment by qzed** [comment]
CC @djrscally

**Comment by djrscally** [comment]
Thanks @qzed.

This is because I broke Surface Go2 support by merging the ov8865 driver updates when the power-control stuff specific to the Go2 wasn't merged...sorry about that. I have a series that can enable it for Go2 users...it won't be what eventually lands upstream (the regulators maintainer wants a less subsystem-specific solution) but merging it here would fix this issue and allow both front and back cameras to work again. 

I just need to rebase it onto linux-surface. I'll work on doing that and open a PR.

Sorry to break things for you @gitwah

**Comment by Lilalas** [comment]
Could the same issue by any chance apply to the SP6 as well? I also couldn’t set up the camera on my device, although I don’t remember the exact output of `cam --list`.

In case this issue is specific to different models, I will open a new issue.

**Comment by djrscally** [comment]
> Could the same issue by any chance apply to the SP6 as well? I also couldn’t set up the camera on my device, although I don’t remember the exact output of `cam --list`.
> 
> In case this issue is specific to different models, I will open a new issue.

No; the power control for that device is the same as the rest. It should be supported already. If you open a new issue, please CC me and include the output of `dmesg` too :)
